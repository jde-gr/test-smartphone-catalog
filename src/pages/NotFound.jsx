import { Link } from 'react-router-dom';

import Layout from '../components/Layout/Layout';
import {
  NotFoundDiv,
  NotFoundMsg,
  NotFoundLink,
} from './styles/NotFound.styles';

const NotFound = () => {
  return (
    <Layout>
      <NotFoundDiv>
        <NotFoundMsg>¡Página no encontrada!</NotFoundMsg>
        <NotFoundLink>
          <Link to={'/'}>
            {'>'} Ir a la página principal {'>'}
          </Link>
        </NotFoundLink>
      </NotFoundDiv>
    </Layout>
  );
};

export default NotFound;
