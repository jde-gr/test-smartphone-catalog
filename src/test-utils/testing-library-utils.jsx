import { render } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from 'react-query';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../store';

const queryClient = new QueryClient();

const AllTheProviders = ({ children }) => {
  return (
    <BrowserRouter>
      <Provider store={store}>
        <QueryClientProvider client={queryClient}>
          {children}
        </QueryClientProvider>
      </Provider>
    </BrowserRouter>
  );
};
const renderWithContext = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { renderWithContext as render };
