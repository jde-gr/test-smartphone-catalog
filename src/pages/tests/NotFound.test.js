import { render } from '../../test-utils/testing-library-utils';
import NotFound from '../NotFound';

test('renders correctly', () => {
  const { container } = render(<NotFound />);
  expect(container).toMatchSnapshot();
});
