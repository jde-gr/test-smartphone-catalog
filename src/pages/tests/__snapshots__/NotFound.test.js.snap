// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="css-1jcmugx"
  >
    <div
      class="css-fi454r"
    >
      <div
        class="css-8bn1o9"
      >
        <img
          alt="logo"
          class="App-logo"
          src="logo.png"
        />
         
        <h1>
          Elige tu móvil
        </h1>
      </div>
      <div
        class="css-y7aohw"
      >
        <a
          href="/"
        >
          <span>
            &gt; 
            Inicio
          </span>
        </a>
      </div>
    </div>
    <div>
      <span
        class="css-cdjgyj"
        data-testid="cartItemQty"
      >
        0
      </span>
    </div>
  </div>
  <main>
    <div
      class="css-1o461mp"
    >
      <h2
        class="css-1mqxd8"
      >
        ¡Página no encontrada!
      </h2>
      <div
        class="css-126uo5m"
      >
        <a
          href="/"
        >
          &gt;
           Ir a la página principal 
          &gt;
        </a>
      </div>
    </div>
  </main>
</div>
`;
